File: TrainingRecommender/Models/UserTraining.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TrainingRecommender.Models
{
    public class UserTraining
    {
        public int Id { get; set; }
        public int Score { get; set; }
        // індекс вправ впливає на кількість вправ (обчислюється за даними користувача)
        public double ExerciseIndex { get; set; }
        public string UserId { get; set; }
        public int TrainingId { get; set; }
        public Training Training { get; set; }
    }
}

File: TrainingRecommender/ClientApp/src/app/models/user-training.ts
import { Training } from './training';

export interface UserTraining {
  id: number;
  score: number;
  // індекс вправ впливає на кількість вправ (обчислюється за даними користувача)
  exerciseIndex: number;
  userId: string;
  trainingId: number;
  training: Training;
}
